clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockInputStream.java

```diff
@@ -142,22 +168,19 @@ private void readBlock() throws IOException {
 
         // verify checksum
         if (flg.isBlockChecksumSet() && Utils.readUnsignedIntLE(in) != checksum.hash(bufferToRead, 0, blockSize, 0)) {
-            throw new IOException(BLOCK_HASH_MISMATCH);
+            throw new LZ4Exception(BLOCK_HASH_MISMATCH);
```

Comment:
I think the approach used by Snappy is a bit better in that it throws a subclass of IOException so that existing IOException catch blocks can catch the exception while allowing more specific catch blocks to exist.